doc/build.rst

```diff
@@ -161,6 +161,24 @@ To build with MinGW, type:
 
   cp make/mingw64.mk config.mk; make -j4
 
+Opening a can of worms - building XGBoost library for python for Windows with MinGW-w64
+---------------------------------------------------------------------------------------
+
+Windows versions of python are built with Microsoft Visual Studio. This has multiple issues:
+
+1. VS is proprietary and commercial software. While Microsoft provides a free edition (Community Edition), installing it may be illegal in your organization.
+2. Visual Studio contains telemetry, as documented in `Microsoft Visual Studio Licensing Terms <https://visualstudio.microsoft.com/license-terms/mt736442/>`_. It `has been spotted on inserting telemetry <https://old.reddit.com/r/cpp/comments/4ibauu/visual_studio_adding_telemetry_function_calls_to/>`_ into apps. At least the Community Edition cannot even be easily downloaded without running the app collecting telemetry. These facts have raised privacy and security concerns among some users and system administrators. Running this kind of software may be against the policy of your organization.
```

Comment:
> At least the Community Edition cannot even be easily downloaded without running the app collecting telemetry

This is redundant, since we stated earlier that VS contains telemetry.

> It `has been spotted on inserting telemetry <https://old.reddit.com/r/cpp/comments/4ibauu/visual_studio_adding_telemetry_function_calls_to/>`_ into apps.

Are we keeping this line?

Reply:
> It has been spotted on inserting telemetry <https://old.reddit.com/r/cpp/comments/4ibauu/visual_studio_adding_telemetry_function_calls_to/>_ into apps.

Let's rephrase this as "Visual Studio has also inserted telemetry into apps for the purpose of instrumentation."

> Running this kind of software may be against the policy of your organization.

Rephrase this as: It may be against the policy of your organization to run software containing telemetry.

In general, we do not want to make strong value judgment on this page. Let's use neutral tone, objectively describe the existence of telemetry, and let users decide whether it's a deal breaker or not.

Reply:
>This is redundant, since we stated earlier that VS contains telemetry.

Not quite.

In order to download VS one has to run an app with telemetry. They provide no offline installer. Instead one runs their app which downloads the msi packages (and there is no easy control on what it downloads and installs, for example I don't need arm toolchain, but it still downloads it) and sends telemetry. So even if one wants to put VS into a sandbox/VM isolated from inet (whkch should prevent leaking telemetry), he will have to create a VM with inet and expect installer in it to send telemetry.

Reply:
Okay, how about "the Visual Studio installer also contains telemetry, requiring internet connection for installation."